frontend/src/pages/modelServing/screens/projects/NIMServiceModal/ManageNIMServingModal.tsx

```diff
@@ -250,15 +272,31 @@ const ManageNIMServingModal: React.FC<ManageNIMServingModalProps> = ({
         ];
 
         if (!editInfo) {
-          if (await isSecretNeeded(namespace, NIM_SECRET_NAME)) {
-            promises.push(
-              createNIMSecret(namespace, NIM_SECRET_NAME, false, false).then(() => undefined),
-            );
-          }
-          if (await isSecretNeeded(namespace, NIM_NGC_SECRET_NAME)) {
-            promises.push(
-              createNIMSecret(namespace, NIM_NGC_SECRET_NAME, true, false).then(() => undefined),
-            );
+          if (constants?.nimSecretName && constants.nimNGCSecretName) {
+            if (await isSecretNeeded(namespace, NIM_SECRET_NAME)) {
+              promises.push(
+                createNIMSecret(
+                  namespace,
+                  NIM_SECRET_NAME,
+                  constants.nimSecretName,
+                  constants.nimNGCSecretName,
+                  false,
+                  false,
+                ).then(() => undefined),
```

Comment:
The NIM_SECRET_NAME here is the name of the secret that we are creating, while the constants.nimSecretName is part of the input used for creating the secret.

Reply:
Looking at it closely, maybe it would be a good idea to simplify the function and make it more maintainable by basing it on the isNGC flag. This flag will dynamically determine the secret's type, name, and data handling logic. Let me know your thoughts.

```
export const createNIMSecret = async (
  projectName: string,
  secretName: string,
  isNGC: boolean,
  dryRun: boolean,
): Promise<SecretKind> => {
  try {
    const data = await getNIMData(secretName, isNGC);

    const newSecret = {
      apiVersion: 'v1',
      kind: 'Secret',
      metadata: {
        name: isNGC ? 'ngc-secret' : 'nvidia-nim-secrets',
        namespace: projectName,
      },
      data,
      type: isNGC ? 'kubernetes.io/dockerconfigjson' : 'Opaque',
    };
```

`await createNIMSecret(projectName, 'nvidia-nim-secrets', false, false);`
`await createNIMSecret(projectName, 'ngc-secret', true, false);`

Reply:
> The NIM_SECRET_NAME here is the name of the secret that we are creating, while the constants.nimSecretName is part of the input used for creating the secret.

Oh gotcha, we are essentially copying the secret from the deployment namespace into the project namespace. So the `secretName` is the destination project secret

Reply:
Updated